### DISCUSSION TITLE
Toolbars at the bottom

### DISCUSSION THREAD

**Discussion by caglarongan** [chrisgurney/obsidian-note-toolbar#37] [Ideas]
Great plugin! It's so important for a plugin to left no trace in notes, this is one of them <3

Can we have an option to place toolbar at the bottom of the page?

- It's too out there when on top, pushing the first heading etc. And if there's also a banner, properties... too much.
- Maybe someone would like to split toolbars depending on the function (one at top, one at bottom).
- It's more accessible for mobile.

**Comment by chrisgurney** [comment]
Thank you!

I am actually considering an enhancement that would let you set the position of each toolbar:

- below the properties/title (default as it is today)
- always at the top (header)
- always at the bottom
- as a floating action button (particularly useful for mobile)

..and you'd be able to select it by platform: desktop, mobile... and possibly by view (reading, editing).

**Comment by caglarongan** [comment]
It's great to hear :)
Thank you for this.

**Comment by chrisgurney** [comment]
@caglarongan 

The [beta release](https://github.com/chrisgurney/obsidian-note-toolbar/discussions/40) now supports toolbar positioning at the top, and lays the foundation for adding bottom positioning later... if I can get it to work cleanly. 

You might also be interested in a command I've added to _completely_ hide the Properties panel, which you could add and toggle from the toolbar if you wanted to.

## Technical Notes

I've done some initial investigation of a bottom toolbar (not a floating action button, yet):

- Regarding the Status bar (desktop):
  - Obsidian places the status bar at the absolute bottom-right position of the overall window, rather than just the editing window. I'm curious where to place this, if a bottom toolbar is added to a note; perhaps try to shift it on top of the toolbar, if one is present?
  - I may have to monitor the height of the toolbar (as they can wrap with lots of items) in order to make sure the status bar is continually in the right spot.
  - Plugins and snippets can effect the appearance (including the height, and potentially the position?) of the status bar as well, so it's possible that snippets may be required for those users, to work around issues I can't accommodate.

- Regarding the nav bar (phone-size screens):
  - Similarly, Obsidian places the nav bar absolutely at the bottom, and its height isn't fixed either, I think requiring a calculation as to where to absolutely place the bottom of the toolbar.
  - I have to monitor for changes to the `mobile-navbar` div, which is `display: none` for `is-tablet`, then comes into view for phone sizes.
  - The `is-tablet` size alone doesn't have a nav bar, so this one's straightforward.

**Reply by caglarongan** [reply]
Oh FINALLY an option to toggle properties visibility in Editor! This is the best solution for this yet, no css, no pseudo hiding. SO useful, I can imagine people would use this plugin even for only that.

I used to use Commander plugin to place "prev/next daily note" commands to status bar, but it was permanent; so I use your plugin now. I love how it's contextual.

In the meantime...
- I scaled down the toolbar's size with css for a more minimal look at the top of the note.
- Made an empty toolbar (named "none") to hide toolbars on my dashboard notes. For example, I don't want the toolbar on my Daily Notes Dashboard; I want it only on the daily notes itself. 
So I add `notetoolbar: none` to properties.
But it's not perfect and still clashes with Banners plugin.
It's maybe a good idea to have an option to hide a folder mapped toolbar per note.


I'm not a dev, so I cannot comment on your technical notes...

**Reply by chrisgurney** [reply]
@caglarongan Thanks so much for sharing! I'd love to hear more about a couple things you mentioned:

1. A screenshot and the CSS you used to scale down the toolbar.

2. What does your folder structure look like that holds your Daily Notes Dashboard and the notes themselves?

**Reply by chrisgurney** [reply]
@caglarongan I also love the idea of a 'none' for the property! I'm going to go ahead and add that to the plugin.

**Reply by chrisgurney** [reply]
@caglarongan The `1.6.0-beta` just released (install via BRAT) has support for setting the the property to `none`, to completely remove the toolbar from that note. Thanks again for the idea! ([Release notes](https://github.com/chrisgurney/obsidian-note-toolbar/releases/tag/1.6.0-beta))

**Comment by caglarongan** [comment]
My folder structure can bee seen on left. I use Folder Notes plugin, so my dashboard note's name is as same as the folder: 01 Daily Notes.md

the css:

```
.callout[data-callout="note-toolbar"] {
scale: .7;
}

```
<img width="1256" alt="SCR-20240423-bsgt" src="https://github.com/chrisgurney/obsidian-note-toolbar/assets/112904909/b2de7ad2-36d3-43f8-8aeb-7702ec31ef3b">

**Reply by chrisgurney** [reply]
Thank you!

- So is your Daily Notes toolbar mapped to the `01 Daily Notes` folder?
  - I ask as I'm looking into supporting a regular expression type folder mapping, which might allow you setup a mapping like this: `01 Daily Notes/[0-9]+` which in theory would completely skip that folder note of yours (so you would not need the `none` property at all).
- Also completely out of curiosity, why did you opt to use `scale` versus `font-size` for that adjustment?

**Reply by chrisgurney** [reply]
@caglarongan Also in `1.6.0-beta` is the ability to override font and icon sizes via the [Style Settings plugin](https://github.com/chrisgurney/obsidian-note-toolbar/wiki/Style-Settings-plugin-support), as inspired by your post. Thank you!

**Reply by caglarongan** [reply]
Great updates, very appreciated !

- Yes, it's mapped to `01 Daily Notes` folder.
- Because I don't know any better :) Not a dev, just using css for obsidian. I thought `scale` might scale down the overall box size and everything in it would follow.

**Comment by chrisgurney** [comment]
@caglarongan (and anyone else reading this):

I'm curious, what do you use this button for on mobile (circled in red)? Do you have it mapped to anything in particular? And do you use it often?

![CleanShot 2024-04-24 at 09 55 42@2x](https://github.com/chrisgurney/obsidian-note-toolbar/assets/17337703/1bd3ade9-122c-46e2-aaba-0759dbe7edd9)

**Reply by caglarongan** [reply]
The mobile app is just a viewer for me. It's mapped to ribbon, but I barely use it.

**Reply by chrisgurney** [reply]
@caglarongan Thanks!

I found out that I can use it to launch a menu that I can put the toolbar's items in. Version 1 of a better mobile experience is in development!

**Comment by chrisgurney** [comment]
@caglarongan In case you missed it, check out the beta for floating buttons and more for mobile!

[1.7 Announcement](https://github.com/chrisgurney/obsidian-note-toolbar/discussions/62)

**Comment by chrisgurney** [comment]
@caglarongan The ability to toggle visibility of icons and labels by platform, floating buttons, and more for mobile is now available in the [1.7 release](https://github.com/chrisgurney/obsidian-note-toolbar/releases/tag/1.7.21), available in Community Plugins.

I may still find a way to stick the toolbar to the bottom of the editor, so I'll leave this feature request open. Thanks again!

**Reply by caglarongan** [reply]
Very appreciated updates, more customization options with Style Settings are just 👌🏻

I disabled beta versions for mobile, it only causes problem for me. So I can only test desktop atm.

Here's the dynamic island for Obsidian 😃 
<img width="955" alt="Screenshot 2024-05-11 at 22 49 23" src="https://github.com/chrisgurney/obsidian-note-toolbar/assets/112904909/7c6b2f08-9dbb-4927-9db2-a5b17bffecf4">

**Comment by chrisgurney** [comment]
Another vote for this from the feedback form:

"Idea for a new toolbar position : same concept as the fixed one at the top, but at the bottom, with a floating design."

**Reply by usernotnull** [reply]
> Another vote for this from the feedback form:
> 
> "Idea for a new toolbar position : same concept as the fixed one at the top, but at the bottom, with a floating design."

Is it on the roadmap?
The plugin is great at making frequently used commands at your fingertips (esp. mobile). On big phones, it's hard to jump too far to the top right, while I could have easily reached it with my normal phone holding position.

My use-case is having the music player controls, and having to press them very often.

**Reply by chrisgurney** [reply]
Hi @usernotnull thanks for your interest in this feature. Yup, it's on the roadmap!

**Comment by chrisgurney** [comment]
@caglarongan @usernotnull `Bottom` position toolbars are now an option [in the beta](https://github.com/chrisgurney/obsidian-note-toolbar/discussions/218). 

Give it a try and let me know how well it works for you!

**Comment by chrisgurney** [comment]
Closing this thread now that it's available in the latest versions.
